src/main/java/com/linecorp/armeria/server/logging/DropwizardMetricCollectingService.java

```diff
@@ -59,13 +86,42 @@
     Function<Service<? super I, ? extends O>, DropwizardMetricCollectingService<I, O>> newDecorator(
             MetricRegistry metricRegistry, String metricNamePrefix) {
 
-        return service -> new DropwizardMetricCollectingService<>(
-                service, metricRegistry, metricNamePrefix);
+        requireNonNull(metricNamePrefix, "metricNamePrefix");
+        return newDecorator(metricRegistry, (ctx, req) -> defaultMetricName(ctx, req, metricNamePrefix));
+    }
+
+    private static String defaultMetricName(
+            ServiceRequestContext ctx, RequestLog req, String metricNamePrefix) {
+
+        String pathAsMetricName = null;
+        String methodName = null;
+
+        if (req.hasAttr(RequestLog.HTTP_HEADERS)) {
+            final HttpHeaders headers = req.attr(RequestLog.HTTP_HEADERS).get();
+            pathAsMetricName = ctx.pathMapping().metricName();
+            methodName = headers.method().name();
+        }
+
+        if (req.hasAttr(RequestLog.RPC_REQUEST)) {
+            methodName = req.attr(RequestLog.RPC_REQUEST).get().method();
+        }
+
+        pathAsMetricName = MoreObjects.firstNonNull(pathAsMetricName, "__UNKNOWN_PATH__");
+
+        if (methodName == null) {
+            methodName = req.method() != null ? req.method() : "__UNKNOWN_METHOD__";
```

Comment:
I think the code before this revision was doing two fallbacks - but now it looks like firstNonNull would work here since __UNKNOWN_METHOD__ is the default.